app/models/miq_expression/field.rb

```diff
@@ -29,11 +30,26 @@ def datetime?
     column_type == :datetime
   end
 
+  def to_s
+    assoc = associations.map { |a| ".#{a}" }.join
```

Comment:
sure feels like this wants to be `associations.join(".")`, but that leading `"."` is tricky.